### PR TITLE
10.9 support

### LINKED ISSUE #143: [BUG]: `System.NullReferenceException: Object reference not set to an instance of an object.`
The plugin doesn't let me sign in. All I get in the UI is an endless spinner (loading animation).
While looking into logs I noticed an error

### Environment
OS: Docker
Jellyfin version: `10.9.0`
Plugin version: `19.2403.7.0` from unstable Repo
Host: Raspberry Pi 5
RAM: 8GB
Storage:
  - Docker - SSD
  - Media  - HDD

The page was reloaded without browser cache and opened in Firefox's Private window to have no cache at all. The issue still persists

<details>
<summary>Log</summary>

```
[18:27:24] [ERR] [15] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL POST /Jellyfin.Plugin.OpenSubtitles/ValidateLoginInfo.

System.NullReferenceException: Object reference not set to an instance of an object.

   at Jellyfin.Plugin.OpenSubtitles.OpenSubtitlesHandler.RequestHandler.SendRequestAsync(String endpoint, HttpMethod method, Object body, Dictionary`2 headers, String apiKey, Int32 attempt, CancellationToken cancellationToken)

   at Jellyfin.Plugin.OpenSubtitles.OpenSubtitlesHandler.OpenSubtitles.LogInAsync(String username, String password, String apiKey, CancellationToken cancellationToken)

   at Jellyfin.Plugin.OpenSubtitles.API.OpenSubtitlesController.ValidateLoginInfo(LoginInfoInput body)

   at lambda_method1005(Closure, Object)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()

--- End of stack trace from previous location ---

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()

--- End of stack trace from previous location ---

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)

   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)

   at Jellyfin.Api.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)

   at Jellyfin.Api.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)

   at Jellyfin.Api.Middleware.IPBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)

   at Jellyfin.Api.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)

   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)

   at Jellyfin.Api.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)

   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)

   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)

   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)

   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)

   at Jellyfin.Api.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)

   at Jellyfin.Api.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)

   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)

   at Jellyfin.Api.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)

   at Jellyfin.Api.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

</details>

Here's what the UI looks like:
image

### PR DESCRIPTION
**Pull request by MBR-0001**
Also "modernized" the code:
- Use file scoped namespaces, which is the reason why diff is big :(
- Use is (not) null
- ~~Bumped .net version to 7~~ lol
- Use PluginServiceRegistrator to register OpenSubtitleDownloader as an ISubtitleProvider because of https://github.com/jellyfin/jellyfin/pull/9191 (that's the only way I know how to do it)

Closes #143

### CODE REVIEW

**Review by crobibero — COMMENTED** [review]
Thank you for making the update, so far it looks great!

**Comment by crobibero** [comment]
If you rebase on the `unstable` branch we can release this into the wild

File: Jellyfin.Plugin.OpenSubtitles/OpenSubtitlesHandler/OpenSubtitlesRequestHelper.cs [hunk 1]
```diff
@@ -10,105 +10,105 @@
...
+            RequestUri = new Uri(url),
+            Content = content,
+            Headers =
             {
-                content = JsonContent.Create(body);
+                Accept = { new MediaTypeWithQualityHeaderValue("*/*") }
             }
+        };
+
+        request.Headers.Add("User-Agent", $"Jellyfin-Plugin-OpenSubtitles v{_version}");
```
Comment (crobibero) [inline Jellyfin.Plugin.OpenSubtitles/OpenSubtitlesHandler/OpenSubtitlesRequestHelper.cs:94]: `request.Headers.UserAgent.Add("Jellyfin-Plugin-OpenSubtitles", _version);` should work

Comment (crobibero) [reply Jellyfin.Plugin.OpenSubtitles/OpenSubtitlesHandler/OpenSubtitlesRequestHelper.cs:94]: @MBR-0001 in case you missed it :)

Comment (MBR-0001) [reply Jellyfin.Plugin.OpenSubtitles/OpenSubtitlesHandler/OpenSubtitlesRequestHelper.cs]: I reverted to just setting it in inside constructor in https://github.com/jellyfin/jellyfin-plugin-opensubtitles/pull/115/commits/996de2db307fe099b3b504ef1bef918861382335

**Comment by MBR-0001** [comment]
I had to do the following modifications to csproj for it to build:
```
<ItemGroup>
    <Reference Include="blah/jellyfin/MediaBrowser.Common/bin/Debug/net8.0/MediaBrowser.Common.dll" />
    <Reference Include="blah/jellyfin/MediaBrowser.Controller/bin/Debug/net8.0/MediaBrowser.Controller.dll" />
    <Reference Include="blah/jellyfin/MediaBrowser.Model/bin/Debug/net8.0/MediaBrowser.Model.dll" />
  </ItemGroup>
```

original is:
```
<PackageReference Include="Jellyfin.Common" Version="10.*-*" />
<PackageReference Include="Jellyfin.Controller" Version="10.*-*" />
```

CI is running so I'm not sure if this builds actually with unchanged csproj

**Review by crobibero — APPROVED** [review]

**Pull request merged**